### PR TITLE
histogram

### PR DESCRIPTION
**Pull request by joaozenobio**
ML research intern application

### CODE REVIEW

**Comment by joaozenobio** [comment]
- [ ] #11814

**Comment by joaozenobio** [comment]
I could not fix the latter error I have commented.

> E           ivy.utils.exceptions.IvyError: tensorflow: histogram: Number of dimensions of `x` and `weights` must coincide. Found: x has 2, weights has 1
> E           Falsifying example: test_histogram(
> E               values=(array([[-1., -1., -1.],
> E                       [-1., -2.,  2.]]),
> E                1,
> E                (0, 1),
> E                True,
> E                True,
> E                'float64',
> E                (-1.0, 1.0),
> E                array([[-1., -1., -1.],
> E                       [-1., -1., -1.]]),
> E                False,
> E                'float64'),
> E               fn_name='histogram',
> E               test_flags=num_positional_args=1. with_out=False. instance_method=False. native_arrays=[False]. container=[False]. as_variable=[False]. test_gradients=False. test_compile=False. ,
> E               ground_truth_backend='tensorflow',
> E               backend_fw=<module 'ivy.functional.backends.torch' from '/ivy/ivy/functional/backends/torch/__init__.py'>,
> E               on_device='cpu',
> E           )
> E           
> E           You can reproduce this example by temporarily adding @reproduce_failure('6.55.0', b'AXicY2BAB4xoNITNCCVQ5UlhMyEZyESBOShsRkYUdzIAAA3sACQ=') as a decorator on your test case

This error appears for TensorFlow when I use multiple axes. It looks like a bug in the Tensorflow function, because, from what I believe, the weights argument shouldn't be modified if I use multiple axes in the histogram function.

**Review by Sam-Armstrong — COMMENTED** [review]
Thanks for the PR! I've had a look at the tests failures, but I can't exactly understand the problem which is causing them. The shape of 'a' and 'weights' both seem to be the same, so not sure why TensorFlow is throwing that ValueError. Perhaps @MuhammadSaeedBatikh can advise on this? Otherwise, I would recommend trying to debug this further yourself, as all the test cases seem to be failing.

**Comment by MuhammadSaeedBatikh** [comment]
> Thanks for the PR! I've had a look at the tests failures, but I can't exactly understand the problem which is causing them. The shape of 'a' and 'weights' both seem to be the same, so not sure why TensorFlow is throwing that ValueError. Perhaps @MuhammadSaeedBatikh can advise on this? Otherwise, I would recommend trying to debug this further yourself, as all the test cases seem to be failing.

Hi @Sam-Armstrong and @joaozenobio,
I ran it again, and the error doesn't seem very helpful. I checked x and w dimensions and they are indeed the same! So, I'm not really sure what the problem is.
 Sorry I couldn't be of help.

File: ivy/functional/backends/torch/experimental/statistical.py [hunk 1]
```diff
@@ -3,10 +3,140 @@
...
+            "bfloat16",
+        )
+    },
+    backend_version,
+)
+def histogram(
+    a: torch.tensor,
+    /,
+    *,
+    bins: Optional[Union[int, torch.tensor, str]] = None,
```
[the remark this answers is not in the document]
Comment (joaozenobio) [reply ivy/functional/backends/torch/experimental/statistical.py]: Yes! I've just corrected it.

**Comment by Sam-Armstrong** [comment]
Thanks for those updates @joaozenobio, but we also need the tests to be passing in order to merge the PR. Would you be able to try debugging the problems further?

**Comment by joaozenobio** [comment]
Hi @Sam-Armstrong, yes. I've raised an issue in GitHub, so now I need to wait for them to answer me.

[tensorflow/tensorflow##60287](https://github.com/tensorflow/tensorflow/issues/60287)

**Comment by Sam-Armstrong** [comment]
Great, sounds good :)

**Comment by joaozenobio** [comment]
Hi @Sam-Armstrong! I've got the response from the Tensorflow Probability repository maintainers and they fixed the problem:

> Fixed in [bcca631](https://github.com/tensorflow/probability/commit/bcca631f01c855425710fe3b8947192b71e310dd)
> 
> Thanks again @joaozenobio

Does the Ivy docker image need to be updated for the tensorflow_probability changes to appear?

**Comment by Sam-Armstrong** [comment]
Hi @joaozenobio, apologies for the slow reply. Thanks for getting that change made in TensorFlow! Have you been able to check locally that your tests now pass with this change made? If they do, I think we can add a comment to your tests stating that they will fail until the next version tensorflow probability version, and go ahead and merge your changes.

**Comment by joaozenobio** [comment]
Hi @Sam-Armstrong! The fix worked under the tf-nightly and tfp-nightly. How should I state the problems at the tests?

```
Python 3.10.9 (main, Mar  1 2023, 18:23:06) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow_probability as tfp
>>> import tensorflow as tf
2023-04-18 23:16:07.988252: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-04-18 23:16:08.431292: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-04-18 23:16:08.432130: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-04-18 23:16:10.041266: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
>>> x = tf.constant([[1, 2, 3], [1, 2, 3.]])
>>> w = tf.constant([[1, 1, 1], [1, 1, 1.]])
>>> e = tf.constant([0, 2.5, 10])
>>> axis = tf.constant([0, 1])
>>> tfp.stats.histogram(x=x, edges=e, axis=axis, weights=w)
<tf.Tensor: shape=(2,), dtype=float32, numpy=array([4., 2.], dtype=float32)>
```

**Comment by Sam-Armstrong** [comment]
@joaozenobio Great! Just put a comment above the test saying something like 'TODO: This test will be fixed by the next tensorflow_probability release, as a fix to the problem has been merged to tfp'. Also maybe give a brief description of the problem/solution in the comment, and include a link to the tfp issue for context. Thanks!

**Comment by Sam-Armstrong** [comment]
Thanks @joaozenobio! Please can you just remove the changes to .idea/vcs.xml from this pull request, then I will get this merged :)

**Comment by joaozenobio** [comment]
Hi @Sam-Armstrong ! I've just discarded the changes from "idea/vcs.xml".

**Comment by Sam-Armstrong** [comment]
Thanks @joaozenobio, great work on this PR! Will get it merged now

**Pull request merged**